User Input - New Design - More Updates

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- https://github.com/google/site-kit-wp/issues/5890#issuecomment-1333528977

## Relevant technical choices

This PR addresses [this comment](https://github.com/google/site-kit-wp/issues/5890#issuecomment-1333402009) and does the following changes:

- Updates the Progress Bar logic so that it is at 75% in the final question, and 100% at the preview screen.
- Removes the initial focus on the first option when a question is unanswered.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

aaemnnosttv commented:
@nfmohit when creating follow-up PRs for the release, please remember to update the target branch as it always defaults to `develop` 👍

aaemnnosttv reviewed (changes requested):
Looks like this PR was created from `develop` which includes many more changes than it should. Would you please recreate this from `main`?

nfmohit commented:
> @nfmohit when creating follow-up PRs for the release, please remember to update the target branch as it always defaults to `develop` 👍

@aaemnnosttv My apologies, I missed it this time. I'll keep it in mind, thank you for correcting it and the heads-up!

> Looks like this PR was created from `develop` which includes many more changes than it should. Would you please recreate this from `main`?

I have rebased this branch and removed the unnecessary changes, leaving only the necessary change. Does it look good now?

Thanks!

aaemnnosttv commented on assets/js/components/user-input/UserInputSelectOptions.js:
```diff
@@ -77,11 +77,6 @@ export default function UserInputSelectOptions( {
 
 		if ( checkedEl ) {
 			focusOption( checkedEl );
-		} else {
-			const el = optionsRef.current.querySelector(
-				`input[type="${ optionType }"]`
-			);
-			focusOption( el );
```
Let's not change this here. It's out of scope for the issue and I'm not sure if we should. Accessibility should probably be reviewed separately once the design iterations are implemented (or nearly done).

nfmohit replied:
Understood, reverted this change. Thank you for confirming, @aaemnnosttv!

aaemnnosttv reviewed (commented):
Thanks for rebasing @nfmohit that's perfectly fine with me.

Let's keep this PR focused on the change to the progress bar and we can revisit a11y as a primary focus in a new issue.

aaemnnosttv reviewed (approved):
Great, thanks @nfmohit!
